### PR TITLE
[ServiceBus] Allow clearing of session state

### LINKED ISSUE #27582: [Service Bus] Session is not closeable
**Describe the bug**
Setting a Service Bus session state to `None` causes an exception

***Exception or Stack Trace***
```
Traceback (most recent call last):
  File "C:/Users/test.py", line 74, in sample_session_send_receive_with_pool
    message_processing(client, queue_name, [])
  File "C:/Users/test.py", line 42, in message_processing
    receiver.session.set_state(None)
  File "C:\Users\venv\lib\site-packages\azure\servicebus\_servicebus_session.py", line 154, in set_state
    MGMT_REQUEST_SESSION_STATE: bytearray(state),
TypeError: cannot convert 'NoneType' object to bytearray
```

**To Reproduce**

 receiver.session.set_state(None)

***Code Snippet***
Add the code snippet that causes the issue.

```
with sb_client.get_queue_receiver(queue_name, max_wait_time=None,
                                              session_id=NEXT_AVAILABLE_SESSION) as receiver:
    renewer = AutoLockRenewer()
    renewer.register(receiver, receiver.session)
    receiver.session.set_state("OPEN")
    for message in receiver.receive_messages(max_message_count=1):
        print(message)
        receiver.complete_message(message)
    receiver.session.set_state(None)
```

**Expected behavior**

Being able to set the state to `None` based on the documentation [Session state remains as long as it isn't cleared up (returning null), even if all messages in a session are consumed.](https://learn.microsoft.com/en-us/azure/service-bus-messaging/message-sessions#message-session-state)

**Screenshots**

**Setup (please complete the following information):**
 - Python Version: Python 3.10
 - SDK Version: azure-servicebus 7.8.1

**Additional context**

Is there another way to close the session because the intended behavior is to close the session and then the receiver fetches another `NEXT_AVAILABLE_SESSION`

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [X] Bug Description Added
- [X] Repro Steps Added
- [X] Setup information Added

### PR DESCRIPTION
**Pull request by kashifkhan**
Per this [document](https://learn.microsoft.com/en-us/azure/service-bus-messaging/message-sessions#message-session-state) `The previously set session state can be cleared by passing null to the SetState method on the receiver.`

The python SDK currently does not accept `None` and raises an Exception since it tries to convert it in to a bytearray.

This brings it in to parity with other languages that accept null, nil etc.

Fixes #27582

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_session.py [hunk 1]
```diff
@@ -122,12 +121,14 @@ def get_state(self, *, timeout: Optional[float] = None, **kwargs: Any) -> bytes:
         session_state = response.get(MGMT_RESPONSE_SESSION_STATE)  # type: ignore
         return session_state
 
-    def set_state(self, state: Union[str, bytes, bytearray], *, timeout: Optional[float] = None, **kwargs: Any) -> None:
+    def set_state(
+        self, state: Union[str, bytes, bytearray, None], *, timeout: Optional[float] = None, **kwargs: Any
+    ) -> None:
```
Comment (kashifkhan) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_session.py:126]: Update for typing changes

File: sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_session_async.py [hunk 2]
```diff
@@ -74,12 +73,12 @@ async def get_state(self, *, timeout: Optional[float] = None, **kwargs: Any) ->
         return session_state
 
     async def set_state(
-        self, state: Union[str, bytes, bytearray], *, timeout: Optional[float] = None, **kwargs: Any
+        self, state: Union[str, bytes, bytearray, None], *, timeout: Optional[float] = None, **kwargs: Any
```
Comment (kashifkhan) [inline sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_session_async.py:76]: Update for typing

File: sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_session_async.py [hunk 3]
```diff
@@ -99,13 +98,13 @@ async def set_state(
...
         state = (
-            state.encode(self._encoding) if isinstance(state, six.text_type) else state
+            state.encode(self._encoding) if isinstance(state, str) else state
         )
         return await self._receiver._mgmt_request_response_with_retry(  # pylint: disable=protected-access
             REQUEST_RESPONSE_SET_SESSION_STATE_OPERATION,
             {
                 MGMT_REQUEST_SESSION_ID: self._session_id,
-                MGMT_REQUEST_SESSION_STATE: bytearray(state),
+                MGMT_REQUEST_SESSION_STATE: bytearray(state) if state else None,
```
Comment (kashifkhan) [inline sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_session_async.py:107]: Actual update to set None, everything else is formatting

**Review by l0lawrence — APPROVED** [review]
lgtm

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_session.py [hunk 4]
```diff
@@ -122,12 +121,14 @@ def get_state(self, *, timeout: Optional[float] = None, **kwargs: Any) -> bytes:
...
-    def set_state(self, state: Union[str, bytes, bytearray], *, timeout: Optional[float] = None, **kwargs: Any) -> None:
+    def set_state(
+        self, state: Optional[Union[str, bytes, bytearray]], *, timeout: Optional[float] = None, **kwargs: Any
+    ) -> None:
         # pylint: disable=protected-access
         """Set the session state.
 
-        :param state: The state value.
-        :type state: Union[str, bytes, bytearray]
+        :param state: The state value. Setting state to None will clear the current session
```
Comment (swathipil) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_session.py:130]: I'm very sorry - baby nit >.<:
```suggestion
        :param state: The state value. Setting state to None will clear the current session.
```

File: sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_session_async.py [hunk 5]
```diff
@@ -74,12 +73,12 @@ async def get_state(self, *, timeout: Optional[float] = None, **kwargs: Any) ->
...
 
     async def set_state(
-        self, state: Union[str, bytes, bytearray], *, timeout: Optional[float] = None, **kwargs: Any
+        self, state: Optional[Union[str, bytes, bytearray]], *, timeout: Optional[float] = None, **kwargs: Any
     ) -> None:
         """Set the session state.
 
-        :param state: The state value.
-        :type state: Union[str, bytes, bytearray]
+        :param state: The state value. Setting state to None will clear the current session
```
Comment (swathipil) [inline sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_session_async.py:80]: baby nit
```suggestion
        :param state: The state value. Setting state to None will clear the current session.
```

**Review by swathipil — APPROVED** [review]
lgtm! (checking that this isn't related to an existing issue/customer report and doesn't need to be linked?)

**Comment by sebastian2696** [comment]
@swathipil this is the issue associated with this #27726

**Comment by swathipil** [comment]
> @swathipil this is the issue associated with this #27726

@sebastian2696 -- looks like you might've accidentally linked back to this PR? 🙂

**Comment by sebastian2696** [comment]
> > @swathipil this is the issue associated with this #27726
> 
> @sebastian2696 -- looks like you might've accidentally linked back to this PR? 🙂

Not that I am aware...

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_session.py [hunk 6]
```diff
@@ -145,14 +146,12 @@ def set_state(self, state: Union[str, bytes, bytearray], *, timeout: Optional[fl
...
-        state = (
-            state.encode(self._encoding) if isinstance(state, six.text_type) else state
-        )
+        state = state.encode(self._encoding) if isinstance(state, str) else state
         return self._receiver._mgmt_request_response_with_retry(  # type: ignore
             REQUEST_RESPONSE_SET_SESSION_STATE_OPERATION,
             {
                 MGMT_REQUEST_SESSION_ID: self._session_id,
-                MGMT_REQUEST_SESSION_STATE: bytearray(state),
+                MGMT_REQUEST_SESSION_STATE: bytearray(state) if state else None,
```
[the remark this answers is not in the document]
Comment (annatisch) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_session.py:154]: Do we need to make a distinction here between `None` and an empty str/bytes? Not sure if the service would treat those the same or not?

File: sdk/servicebus/azure-servicebus/tests/async_tests/test_sessions_async.py [hunk 7]
```diff
@@ -874,6 +874,9 @@ async def test_async_session_get_set_state_with_receiver(self, servicebus_namesp
                     count += 1
                 state = await receiver.session.get_state()
                 assert state == b'first_state'
+                await receiver.session.set_state(None, timeout=5)
+                state = await receiver.session.get_state()
+                assert not state
```
Comment (annatisch) [inline sdk/servicebus/azure-servicebus/tests/async_tests/test_sessions_async.py:879]: Would also be good to clarify if a distinction between None and an empty string should be made here too.

**Pull request merged**